Realm/RLMApp.mm

```diff
@@ -128,21 +134,35 @@ - (instancetype)initWithBaseURL:(nullable NSString *)baseURL
         self.localAppName = localAppName;
         self.localAppVersion = localAppVersion;
         self.defaultRequestTimeoutMS = defaultRequestTimeoutMS;
-
-        _config.device_info.sdk = "Realm Swift";
-
-        // Platform info isn't available when running via `swift test`.
-        // Non-Xcode SPM builds can't build for anything but macOS, so this is
-        // probably unimportant for now and we can just report "unknown"
-        auto processInfo = [NSProcessInfo processInfo];
-        RLMNSStringToStdString(_config.device_info.platform_version,
-                               [processInfo operatingSystemVersionString] ?: @"unknown");
-        RLMNSStringToStdString(_config.device_info.sdk_version, REALM_COCOA_VERSION);
+        RLMConfigureSyncConnectionParameters(_config);
         return self;
     }
     return nil;
 }
 
+static void RLMConfigureSyncConnectionParameters(realm::app::App::Config config) {
```

Comment:
This function doesn't actually do anything. It needs to take a `realm::app::App::Config&` so that it's not just mutating a local variable.

This is now doing some not totally trivial things, so it'd be nice to have a test. I don't think there's a _great_ way to test it, but maybe you could do something like override the logger before opening a synchronized Realm and verifying that all of the expected values are present in the logs.

Reply:
I added a test for this, using the `Logger`, while on it I found a bug which is fixed on this PR